### PR TITLE
Pass down KnownDeployProperties to the GenerateCreateScript functionality

### PR DESCRIPTION
**Pull request by dovic95**
This PR allows setting all the possible properties in the `GenerateCreateScript` functionality. The properties defined in the `csproj` (the `KnownDeployProperties` ones) are now automatically set when generating the create script

Solves issue #230

### CODE REVIEW

File: src/DacpacTool/PackageBuilder.cs
Comment (jmezach) [inline src/DacpacTool/PackageBuilder.cs:392]: I'd like to keep the seperation of concerns here and not pass in the `BuildOptions` directly, but keep the seperate parameters. I guess we could just replace the boolean with DacDeployOptions.

File: src/DacpacTool/PackageDeployer.cs
Comment (jmezach) [inline src/DacpacTool/PackageDeployer.cs:216]: Can't we replace this method with the `PropertyParser` to avoid duplication of logic?

File: src/DacpacTool/BuildOptions.cs
Comment (jmezach) [inline src/DacpacTool/BuildOptions.cs:15]: I'm a bit on the fence about the name of this property. Technically they are used when generating the create script, but essentially they are deployment options, so perhaps `DeployProperty` would be better name? And then rename `Property` to `BuildProperty` as well, to make the distinction clear?

File: test/DacpacTool.Tests/PackageBuilderTests.cs
Comment (jmezach) [inline test/DacpacTool.Tests/PackageBuilderTests.cs:644]: It feels a bit weird to have these seperate test cases for what is essentially the same logic, which is making sure we can pass along deployment properties.

File: test/DacpacTool.Tests/PropertyParserTest.cs
Comment (jmezach) [inline test/DacpacTool.Tests/PropertyParserTest.cs:1]: Please remove this copyright notice.

File: src/DacpacTool/PropertyParser.cs
Comment (jmezach) [inline src/DacpacTool/PropertyParser.cs:1]: Please remove this useless newline.

**Review by jmezach — CHANGES_REQUESTED** [review]
@dovic95 Thank you for your contribution. I have a couple of comments.

File: src/DacpacTool/BuildOptions.cs
Comment (dovic95) [reply src/DacpacTool/BuildOptions.cs]: I've renamed the properties as suggested

File: src/DacpacTool/PackageBuilder.cs
Comment (dovic95) [reply src/DacpacTool/PackageBuilder.cs]: Done. I agree it's better to separate the two concepts and apply the Hollywood principle :)

File: src/DacpacTool/PackageDeployer.cs
Comment (dovic95) [reply src/DacpacTool/PackageDeployer.cs]: Done

File: test/DacpacTool.Tests/PropertyParserTest.cs
Comment (dovic95) [reply test/DacpacTool.Tests/PropertyParserTest.cs]: Removed

File: test/DacpacTool.Tests/PackageBuilderTests.cs
Comment (dovic95) [reply test/DacpacTool.Tests/PackageBuilderTests.cs:638]: I can remove these tests but I think we should be 100% sure the deployment options are not modified inside the `GenerateCreateScript` method.

I can add a test where I first generate the script using `DacServices.GenerateCreateScript` method and then compare it to the one generated by the `GenerateCreateScript` method.

File: src/DacpacTool/PropertyParser.cs
Comment (jmezach) [inline src/DacpacTool/PropertyParser.cs:30]: `ToDacDeployOptions` suggests we are converting one into the other, while actually we're just extracting a subset of the `BuildOptions` instance into a `DacDeployOptions` object. Perhaps `ExtractDeployOptions` would be a more appropriate name?

File: test/DacpacTool.Tests/PackageBuilderTests.cs
Comment (jmezach) [reply test/DacpacTool.Tests/PackageBuilderTests.cs:638]: Yeah, I think that makes sense. My comment was more about that we now have 2 separate tests that trigger the same logic, but the only difference is setting a different property on the `DacDeployOptions` class.

File: src/DacpacTool/PackageDeployer.cs
Comment (jmezach) [inline src/DacpacTool/PackageDeployer.cs:198]: Is it just me, or does it still feel like we're duplicating some logic here that's also in the `PropertyParser` class now?

File: test/DacpacTool.Tests/PackageBuilderTests.cs
Comment (dovic95) [reply test/DacpacTool.Tests/PackageBuilderTests.cs]: I've removed the two tests and added one that checks the expected create script

Comment (jmezach) [reply test/DacpacTool.Tests/PackageBuilderTests.cs]: Looks good :)

File: src/DacpacTool/PackageDeployer.cs
Comment (dovic95) [reply src/DacpacTool/PackageDeployer.cs:198]: Where do you see the duplication? The `foreach` ?

Comment (dovic95) [reply src/DacpacTool/PackageDeployer.cs:198]: Mmmh we could actually have only one method for setting properties in a `DacDeployOptions` object. I just have a doubt regarding this verification and if it's applicable during the build:

```
                if (databaseProperty.Name == "SqlCommandVariableValues")
                {
                    throw new ArgumentException("SQLCMD variables should be set using the --sqlcmdvar command line argument and not as a property.");
                }
```

I'll have a look in a few hours

Comment (dovic95) [reply src/DacpacTool/PackageDeployer.cs]: I've mutualized the `SetDeployProperties` method

File: src/DacpacTool/PropertyParser.cs
Comment (dovic95) [reply src/DacpacTool/PropertyParser.cs]: I've renamed the method.

**Comment by dovic95** [comment]
@jmezach I've resolved all the comments. Let me know if the PR is OK for you

File: src/DacpacTool/PropertyParser.cs
Comment (jmezach) [reply src/DacpacTool/PropertyParser.cs]: It seems this newline is still here?

Comment (dovic95) [reply src/DacpacTool/PropertyParser.cs]: I don't see it on my side:
image

Comment (jmezach) [reply src/DacpacTool/PropertyParser.cs]: Nevermind, I wasn't looking straight ;). Must be the end of the year.

**Comment by jmezach** [comment]
@dovic95 Looks good to me.

@ErikEJ @jeffrosenberg Could either of you have a look at this as well, thanks!

**Comment by ErikEJ** [comment]
LGTM

**Review by jmezach — APPROVED** [review]

**Comment by dovic95** [comment]
Hi, are we waiting for another approval? Would be great if we can complete this PR and release the version :)

**Review by jeffrosenberg — APPROVED** [review]
LGTM

**Pull request merged**
